web/web.go

```diff
@@ -0,0 +1,771 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Package web defines web APIs to be used with harbourbridge frontend.
+// Apart from schema conversion, this package involves API to update
+// converted schema.
+
+package web
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+	"time"
+
+	//"harbourbridge-web/models"
+	"github.com/cloudspannerecosystem/harbourbridge/conversion"
+	"github.com/cloudspannerecosystem/harbourbridge/internal"
+	"github.com/cloudspannerecosystem/harbourbridge/mysql"
+	"github.com/cloudspannerecosystem/harbourbridge/postgres"
+	"github.com/cloudspannerecosystem/harbourbridge/spanner/ddl"
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/gorilla/handlers"
+	_ "github.com/lib/pq"
+)
+
+// TODO(searce):
+// 1) Test cases for APIs
+// 2) API for saving/updating table-level changes.
+// 3) API for showing logs
+// 4) Split all routing to an route.go file
+// 5) API for downloading the schema file, ddl file and summary report file.
+// 6) Update schema conv after setting global datatypes and return conv. (setTypeMap)
+// 7) Add rateConversion() in schema conversion, ddl and report APIs.
+// 8) Add an overview in summary report API
+var mysqlTypeMap = make(map[string][]typeIssue)
+var postgresTypeMap = make(map[string][]typeIssue)
+
+func homeLink(w http.ResponseWriter, r *http.Request) {
+	fmt.Fprintf(w, "Welcome to Harbourbridge!")
+	w.WriteHeader(http.StatusOK)
+}
+
+// Driver config is used for direct database connection.
+type DriverConfig struct {
```

Comment:
DriverConfig seems to only be used in the web package -- make it private?

Reply:
maybe we should make it private and add comment as mentioned in option 3 as well

Reply:
made driverConfig private and added comment on top